extractor/src/main/java/org/schabi/newpipe/extractor/stream/Stream.java

```diff
@@ -1,108 +1,253 @@
+     * @param streamList the list of {@link Stream Streams} which will be compared
+     * @return whether the list already contains one stream with equals stats
      */
-    public Stream(String url, String torrentUrl, MediaFormat format) {
-        this.url = url;
-        this.torrentUrl = torrentUrl;
-        this.format = format.id;
-        this.mediaFormat = format;
+    public static boolean containSimilarStream(final Stream stream,
+                                               final List<? extends Stream> streamList) {
+        if (isNullOrEmpty(streamList)) {
+            return false;
+        }
+        for (final Stream cmpStream : streamList) {
+            if (stream.equalStats(cmpStream)) {
+                return true;
+            }
+        }
+        return false;
     }
 
     /**
-     * Reveals whether two streams have the same stats (format and bitrate, for example)
+     * Reveals whether two streams have the same stats ({@link MediaFormat media format} and
+     * {@link DeliveryMethod delivery method}).
+     *
+     * <p>
+     * If the {@link MediaFormat media format} of the stream is unknown, the streams are compared
+     * by only using the {@link DeliveryMethod delivery method} and their id.
+     * </p>
+     *
+     * <p>
+     * Note: This method always returns always false if the stream passed is null.
+     * </p>
+     *
+     * @param cmp the stream object to be compared to this stream object
+     * @return whether the stream have the same stats or not, based on the criteria above
      */
-    public boolean equalStats(Stream cmp) {
-        return cmp != null && getFormatId() == cmp.getFormatId();
+    public boolean equalStats(@Nullable final Stream cmp) {
+        if (cmp == null) {
+            return false;
+        }
+
+        Boolean haveSameMediaFormatId = null;
+        if (mediaFormat != null && cmp.mediaFormat != null) {
+            haveSameMediaFormatId = mediaFormat.id == cmp.mediaFormat.id;
+        }
+        final boolean areUsingSameDeliveryMethodAndAreUrlStreams =
+                deliveryMethod == cmp.deliveryMethod && isUrl == cmp.isUrl;
+
+        return haveSameMediaFormatId != null
+                ? haveSameMediaFormatId && areUsingSameDeliveryMethodAndAreUrlStreams
+                : areUsingSameDeliveryMethodAndAreUrlStreams;
     }
 
     /**
-     * Reveals whether two Streams are equal
+     * Reveals whether two streams are equal.
+     *
+     * @param cmp the stream object to be compared to this stream object
+     * @return whether streams are equal
      */
-    public boolean equals(Stream cmp) {
-        return equalStats(cmp) && url.equals(cmp.url);
+    public boolean equals(final Stream cmp) {
+        return equalStats(cmp) && content.equals(cmp.content);
     }
 
     /**
-     * Check if the list already contains one stream with equals stats
+     * Gets the identifier of this stream, e.g. the itag for YouTube.
```

Comment:
Is it really needed to add `This method`, because we know that's a method and a lot of methods JavaDocs doesn't include this, at least on the first sentence?